open-craft/xblock-dalite
# Studio edit and lms

**Description:** First usable version of Dalite Xblock.
**JIRA:** [OC-1537](https://tasks.opencraft.com/browse/OC-1537)

**Testing instructions:**
1. Set up [dalite](https://github.com/open-craft/dalite-ng).
2. Create an assignment and a couple of questions.
3. Install this XBlock.
4. Create a course.
5. Add `xblock-dalite` to `Advanced Modules List`
6. Add dalite-ng LTI passport. _Note:_ Dalite-xblock introduces different LTI passport format: `(dalite-xblock)[LTI_ID];[dalite_root_url];[LTI_KEY];[LTI_SECRET]`:
6.1. `(dalite-xblock)` is a prefix denoting Dalite-XBlock LTI Passport
6.2. `LTI_ID` is just an identifier - it is displayed to course author in Dalite-XBlock edit dialog.
6.3. `dalite_root_url` is Dalite-ng root URL (don't add `/lti/`here). Should be tolerant to trailing slash.
6.4. LTI_KEY and LTI_SECRET have the same meaning as for ordinary LTI passport.
6.5. **Most important change:** since `dalite_root_url` contains at least one colon for sure (`http:` or `https:`), separator is changed to semicolon `;`.
7. (Optinal) Add a couple of Dalite-XBlock LTI Passports more (i.e. all the same, except different LTI ID)
8. Go to Outline -> Add Section, subsection and unit -> open unit for editing.
9. "Advanced" components should be available and contain "Dalite XBlock".
10. Add Dalite XBlock.
11. Click Edit.
12. Fill in Assignment ID and Question ID with ID of assignment and question you created at step 2.
13. _Verify:_ LTI ID is a dropdown and is pre-popuilated with first Dalite-XBlock LTI Passport.
14. (Optional) _Verify:_ LTI ID should display all Dalite-XBlock LTI Passports configured in steps 6 and 7.
15. Don't edit anything else (even LTI ID field) - click Save.
16. _Verify:_ Dalite-ng is shown in an iframe, with target assignment and question opened.
17. Publish unit.
18. Log in to LMS as student (i.e. audit or honor)
19. Enroll to a course (if not already enrolled), open unit with Dalite XBlock in it.
20. _Verify:_ Dalite-ng is shown in an iframe with target assignment and question opened.
21. _Verify:_ dalite-ng workflow works: can submit answer and rationale, view other rationales and stick with your answer or change answer.
